### DISCUSSION TITLE
Do I need to `useShallow` for object-containing properties?

### DISCUSSION THREAD

**Discussion by MonstraG** [pmndrs/zustand#2815] [Q&A] [answered]
Store:
```tsx
export const useTimerStore = create({
	clock: 0,
	state: "paused",
	settings: {
		duration: 0,
		color: "blue"
	}
});
```

`settings`, as you can see, is an object.

Component:
```tsx
const settingsSelector = (store) => store.settings

const MyComponent = () => {
	const settings = useTimerStore(settingsSelector);
	
	return ...stuff to view and change settings
}
```

Both https://zustand.docs.pmnd.rs/guides/prevent-rerenders-with-use-shallow and https://zustand.docs.pmnd.rs/migrations/migrating-to-v5#requiring-stable-selector-outputs have examples where
1) selector function is inside the render
2) object (or array) is constructed "on the fly" and isn't part of the store. 

So my question is: is this a stable selector output or not?

**Comment by dbritto-dev** [comment, marked as answer]
@MonstraG it's stable selector result
